lib/uplinkc/object.go

```diff
@@ -65,7 +135,7 @@ func upload_write(uploader C.UploaderRef, bytes *C.uint8_t, length C.int, cErr *
 
 	n, err := upload.wc.Write(buf)
 	if err == io.EOF {
-		return C.EOF
+		return C.int(0)
```

Comment:
Shouldn't it return `n` instead of `0`? `n` should be the number of bytes successfully written before the error occurred.

Reply:
Shouldn't we also set `cErr` to `err`, if `err != nil && err != io.EOF`?

Reply:
yea - just talked with alex about it and we arrived at the same conclusion

Reply:
good catch